packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -97,8 +98,8 @@ class SocksProxyConnection {
   }
 
   async connect() {
-    if (this.socksProxy.proxyAgentFromOptions)
-      this.target = await this.socksProxy.proxyAgentFromOptions.connect(new EventEmitter() as any, { host: rewriteToLocalhostIfNeeded(this.host), port: this.port, secureEndpoint: false });
+    if (this.socksProxy.proxyAgentFromOptions instanceof SocksProxyAgent)
```

Comment:
Looks wrong, it can be also a HttpsProxyAgent.

Reply:
In a file called `socksClientCertificatesInterceptor`?

Reply:
Yes.

Reply:
Background: Only the communication between the browser and Playwright is made via a Socks Proxy, hence its called like that. Internally when we make requests to the target (e.g. internal.corp.com it can be either direct or via an agent. An agent is a proxy agent which then can be https or socks.). (We don't intercept http traffic.)